src/libs/actions/Report.js

```diff
@@ -896,6 +896,29 @@ NetworkConnection.onReconnect(() => {
     fetchAll(false);
 });
 
+function editReportComment(reportID, reportAction, htmlForNewComment) {
+    // Optimistically update the report action with the new message
+    const sequenceNumber = reportAction.sequenceNumber;
```

Comment:
I was just reviewing @sketchydroide's [delete comment PR](https://github.com/Expensify/Expensify.cash/pull/2046), and I thought the way he approached this syntactically was a bit cleaner (and has better error handling):

```javascript
function editReportComment(reportID, reportAction, htmlForNewComment) {
    // Optimistically update Onyx
    actionToMerge = {}'
    const oldMessage = {...reportAction.message};
    actionToMerge[reportAction.sequenceNumber] = {
        ...reportAction,
        message: [
            {
                type: 'COMMENT',
                html: htmlForNewComment,
                text: htmlForNewComment.replace(/<[^>]*>?/gm, ''),
            },
        ],
    };
    
    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, actionToMerge);
    
    // Make the API request
    API.Report_EditComment({
        reportID,
        reportActionID: reportAction.reportActionID,
        reportComment: htmlForNewComment,
    })
        .catch(() => {
            // If it fails, reset Onyx
            actionToMerge[reportAction.sequenceNumber] = {
                ...reportAction,
                message: oldMessage,
            };

            Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, reportActionsToMerge);
        });
}
```